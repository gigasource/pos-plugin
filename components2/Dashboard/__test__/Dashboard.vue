<script>
import ContentRender from "../../../components/common/ContentRender";
import {KeepAlive, onActivated, ref, h} from "vue";
import {contentView} from "./dashboard-shared";
import _ from 'lodash';


const TextField = {
  setup(props) {
    const count = ref(0);
    const init = _.debounce(() => {
      count.value ++;
    }, 10);
    init();
    onActivated(() => {
      init();
      console.log('onActivated');
    })

    return () => <p>{count.value}</p>
  }
}

const TextComp = function () {
  return <p>c</p>
}

const views = {
  TextComp,
  TextField
}

export default {
  name: "Dashboard",
  components: {
    ContentRender,
  },
  setup() {
    return () => <KeepAlive>
      {h(views[contentView.value.name])}
    </KeepAlive>
  }
}
</script>

<style scoped>

</style>
