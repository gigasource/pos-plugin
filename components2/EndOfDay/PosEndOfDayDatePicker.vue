<script>
import {PosEndOfDayDatePicker} from "./PosEndOfDayDatePickerUi";

export default PosEndOfDayDatePicker;
</script>

<style lang="scss" scoped>
/*@import "../../style/variables";*/
/*@import "../../style/colors";*/
.g-picker {
  background-color: #eee;
}

button {
  outline: none;
}

:deep(table) {
  border-collapse: collapse;
}

$textDisabled: #9e9e9e;
$primary-transition: 0.5s cubic-bezier(0.25, 0.8, 0.5, 1) !default;

/*TITLE*/
.g-date-picker-title {
  display: flex;
  flex-direction: column;
  line-height: 1;

  &__year {
    display: inline-flex;
    font-size: 14px;
    font-weight: 500;
    margin-bottom: 8px;
    cursor: pointer;
  }

  &__date {
    font-size: 24px;
    text-align: left;
    font-weight: 500;
    position: relative;
    overflow: hidden;
    padding-bottom: 8px;
    margin-bottom: -8px;
  }

  &--disabled {
    color: $textDisabled;
  }
}

/*YEAR PICKER*/
.g-date-picker-years {
  font-size: 16px;
  font-weight: 400;
  height: 50vh;
  list-style-type: none;
  overflow: auto;
  text-align: center;
  padding: 0;
  margin-top: 0;
  margin-bottom: 0;

  li {
    cursor: pointer;
    padding: 8px 0;
    transition: none;
    width: 100%;
    display: inline-block;

    &.active {
      font-size: 26px;
      font-weight: 500;
      padding: 10px 0;
    }

    &:hover {
      background: rgba(0, 0, 0, 0.12);
    }
  }
}

/*HEADER*/
:deep(.g-date-picker-header) {
  padding: 8px 0 8px 0;
  margin-left: 12px;
  margin-right: 12px;
  height: 50px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: relative;
  background-color: transparent;

  .g-date-picker-header-container {
    width: 27%;
    margin-right: auto;
    margin-left: auto;
    display: flex;

    .g-date-picker-header__value {
      flex: 1;
      position: relative;

      div {
        transition: $primary-transition;
        width: 100%;
        display: flex;
        justify-content: center;
      }

      button {
        font-size: 18px;
        font-style: normal;
        cursor: pointer;
        font-weight: bold;
        outline: none;
        padding: 0.5rem;
        transition: $primary-transition;
        background-color: transparent;
        border-style: none;
        color: #212B35;
      }

      .g-date-picker-header--disabled {
        button {
          color: $textDisabled;
        }
      }
    }

    %navigate_button {
      width: 24px;
      padding: 0;
      border: none;
      font-size: 25px;
      background-color: transparent;
    }

    .g-date-picker-header__prev-button {
      @extend %navigate_button;

      &::after {
        content: '\F141';
        font-family: "Material Design Icons";
      }
    }

    .g-date-picker-header__next-button {
      @extend %navigate_button;

      &::after {
        content: '\F142';
        font-family: "Material Design Icons";
      }
    }
  }
}

/*TABLE*/
:deep(.g-date-picker-table) {
  position: relative;
  padding: 0 16px;
  height: 100%;
  background-color: #eeeeee;

  table {
    transition: $primary-transition;
    table-layout: fixed;
    width: 100%;

    thead {
      tr:first-child {
        border-left: 1px solid #E0E0E0;
      }

      tr:last-child {
        border-right: 1px solid #E0E0E0;
      }
    }
  }

  tr {
    height: 32px;
  }

  td, th {
    text-align: center;
    position: relative;
  }

  th {
    background-color: #FFFFFF;
    font-size: 13px;
    font-weight: normal;
    line-height: 17px;
    color: #6e6e6e;
    border-top: 1px solid #E0E0E0;
  }

  td {
    height: 13.4vh;
    border: 1px solid #E0E0E0;
    position: relative;
    color: #9E9E9E;

    button .g-table-item__content {
      font-family: Muli;
      font-size: 20px;
      font-weight: bold;
    }
  }

  button {
    background-color: transparent;
    border-style: none;
    color: inherit;
    box-shadow: none;
  }

  .g-date-picker-table--date .g-table-item {
    height: 32px;
    width: auto;
  }

  .g-table-item {
    margin: 0;
    z-index: auto;
    font-size: 12px;
    background-color: #ffffff;
    border-style: none;
    color: inherit;
    width: 100%;
    height: 100%;

    &--active {
      color: #9E9E9E;
      border: 2px solid #1271FF !important;
    }

    &--start-range, &--end-range {
      color: #fff;
      text-shadow: 0 4px 4px rgba(0, 0, 0, 0.25);
    }

    &--in-range {
      color: #232323;
    }

    &--outlined {
      border: thin solid #232323;
    }

    &--readonly {
      pointer-events: none;
    }

    &--disabled {
      color: #E0E0E0;
      border-color: $textDisabled;
    }

    &__background {
      width: 0;
      top: 2px;
      height: 32px;
      position: absolute;
      z-index: -1;

      &--start-range {
        width: 50%;
        right: 0;
      }

      &--in-range {
        width: 100%;
        right: 0;
      }

      &--end-range {
        width: 50%;
        left: 0;
      }
    }
  }
}

.g-date-picker-table--date {
  th {
    padding: 8px 0;
    font-weight: 600;
    color: rgba(0, 0, 0, 0.38);
  }

  td {
    width: 32px;
    padding: 0;
  }

  & .g-date-picker-table__events {
    bottom: 6px;
    height: 8px;
    left: 0;
    position: absolute;
    text-align: center;
    white-space: pre;
    width: 100%;

    > div {
      border-radius: 50%;
      display: inline-block;
      height: 8px;
      margin: 0 1px;
      width: 8px;
    }
  }
}

.g-date-picker-table--month {
  td {
    width: 33.333333%;
    height: 15vh;

    .g-table-item {
      width: 100%;
      height: 100%;
    }
  }
}
</style>
