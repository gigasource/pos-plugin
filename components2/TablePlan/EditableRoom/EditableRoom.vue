<script>

import RoomFactory from '../RoomFactory'
import { computed, watch } from 'vue'
import EditableRoomEventHandlersFactory from './EventHandlersForEditableRoom'
import { globalZoom, isSelectingObject } from '../../View/EditTablePlan/EditTablePlanLogics';
import { getScopeAttrs } from '../../../utils/helpers';
import RoomStyleFactory from '../RoomStyles';
import Touch from '../../../../../../pos-vue-framework/src/directives/touch/touch'
import { isTable } from '../room-logic';

const { hooks, fn } = RoomFactory()

const { roomObjectContainerStyle, roomObjectStyle } = RoomStyleFactory()

const { resizeRoomObjectEvenHandler: e, roomContainerEventHandlers:r, touchHandlers:t } = EditableRoomEventHandlersFactory()

hooks.on('provideZoom', ({ zoom }) => {
  watch(() => zoom.value, () => {
    globalZoom.value = zoom.value
  })
})
const Component = fn()
const _roomObjectContainerStyle = (obj) => {
  const style = roomObjectContainerStyle(obj)
  if (isSelectingObject(obj)) {
    style.border = '1px solid #1271FF'
    style.zIndex = 50
  }
  return style
}

export default {
  directives: {Touch},
  setup() {

    const roomObjectRenderFn = (obj) => <div style={roomObjectStyle(obj)}>

      {isTable(obj) ? obj.name : ''}
    </div>

    const style = computed(() => ({
      width: `${globalZoom.value * 20}px`,
      height: `${globalZoom.value * 20}px`,
      'border-radius': '100%',
      'background-color': '#2979FF',
      position: 'absolute',
      right: 0,
      bottom: 0,
      transform: `translate(100%, 100%)`,
      display: 'flex',
      'justify-content': 'center',
      'align-items': 'center',
      zIndex: 100
    }))

    const arrowStyle = computed(() => ({
      width: `${13 * globalZoom.value}px`,
      height: `${13 * globalZoom.value}px`,
    }))

    const resizeButtonRenderFn = obj => isSelectingObject(obj) ?
        <div style={style.value} {...getScopeAttrs()}
             {...e(obj)}
        >
          <img alt style={arrowStyle.value} src="/plugins/pos-plugin/assets/resize.svg" draggable="false"/>
        </div>
        : null
    return () => <Component v-slots={
      {
        'room-object': (obj) => {
          return <div key={obj._id} id={obj.name}
                      v-touch={t(obj)}
                      style={_roomObjectContainerStyle(obj)}
          >
            {roomObjectRenderFn(obj)}
            {resizeButtonRenderFn(obj, getScopeAttrs())}
          </div>
        }
      }
    }> </Component>
  }
}
</script>
<style scoped lang="scss">
.room {
  position: relative;
  width: 100%;
  height: 100%;
  background-image: url('/plugins/pos-plugin/assets/out.png');

  &__object__rotate {
    width: 16px;
    height: 16px;
    border-radius: 100%;
    background-color: #2979FF;
    position: absolute;
    right: 0;
    bottom: 0;
    transform: translateX(100%);
    display: flex;
    justify-content: center;
    align-items: center;
  }
}
</style>
