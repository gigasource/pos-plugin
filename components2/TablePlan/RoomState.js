import { computed, ref } from 'vue';
import _ from 'lodash';
import { createRoom } from './RoomLogics';
import cms from 'cms';

export const roomsStates = ref([]);

export const fetchRooms = async function () {
  let _rooms = await cms.getModel('Room').find({})
  _rooms = _.sortBy(_rooms.map(r => createRoom(r)), r => r.room.order)
  roomsStates.value = _rooms;
}

export const addRoom = async function (room) {
  const createdRoom = await cms.getModel('Room').create(room)
  roomsStates.value.push(createRoom(createdRoom))
  return createdRoom
}

export const removeRoom = async function (room) {
  const idx = _.findIndex(roomsStates.value, r => r.room._id.toString() ===room._id.toString())
  if (idx !== -1) {
    roomsStates.value.splice(idx, 1)
    await cms.getModel('Room').remove({ _id: room._id })
  }
}

//w

export const removeSelectingRoom = async function() {
  await removeRoom(selectingRoomStates.value.room)
  //todo: auto change selecting room
  if (roomsStates.value.length > 0) {
    selectingRoomStates.value = roomsStates.value[0]
  }
}
export const selectingRoomStates = ref(null);

export const onSelectRoom = function (newSelectingRoom) {
  selectingRoomStates.value = _.find(roomsStates.value, r => r.room._id.toString() === newSelectingRoom.room._id)
}

export const objectsInSelectingRoom = computed(() => (selectingRoomStates.value) ? selectingRoomStates.value.room.roomObjects : [])

const roomsName = computed(() => {
  return roomsStates.value.map(r => r.room.name)
})


export const newRoomName = computed(() => {
  let res = 1
  while (roomsName.value.includes('' + res)) res++
  return "" + res
})

export const nextOrder = computed(() => {
  return (_.maxBy(roomsStates.value, r => r.room.order).room.order + 1)
})

export const addNewRoom = async function() {
  const newName = newRoomName.value

  const newRoom = {name: newName , order: nextOrder.value}
  await addRoom(newRoom)
}

export const updateRoomName = async function(room) {
  await cms.getModel('Room').findOneAndUpdate({_id: room._id}, {$set: {name: room.name}})
}

export const updateSelectingRoomName = async function(newName) {
  selectingRoomStates.value.room.name = newName
  await updateRoomName(selectingRoomStates.value.room)
}


export const swapSelectingRoomOrderWithTheRoomBefore = async function() {
  //todo: check if is selecting an object or not
  const idx = _.findIndex(roomsStates.value, r => r.room._id === selectingRoomStates.value.room._id)
  if (idx > 0) {
    const order = (await cms.getModel('Room').findOne({_id: roomsStates.value[idx - 1].room._id})).order
    const order1 = (await cms.getModel('Room').findOne({_id: roomsStates.value[idx].room._id})).order
    await cms.getModel('Room').findOneAndUpdate({_id: roomsStates.value[idx - 1].room._id}, {$set: {order: order1}})
    await cms.getModel('Room').findOneAndUpdate({_id: roomsStates.value[idx].room._id}, {$set: {order: order}})
    await fetchRooms()
  }
}

export const swapSelectingRoomOrderWithTheRoomBehind = async function() {
  const idx = _.findIndex(roomsStates.value, r => r.room._id === selectingRoomStates.value.room._id)
  if (idx < roomsStates.value.length - 1 && idx !== -1) {
    const order = (await cms.getModel('Room').findOne({_id: roomsStates.value[idx].room._id})).order
    const order1 = (await cms.getModel('Room').findOne({_id: roomsStates.value[idx + 1].room._id})).order
    await cms.getModel('Room').findOneAndUpdate({_id: roomsStates.value[idx].room._id}, {$set: {order: order1}})
    await cms.getModel('Room').findOneAndUpdate({_id: roomsStates.value[idx + 1].room._id}, {$set: {order: order}})
    await fetchRooms()
  }
}
