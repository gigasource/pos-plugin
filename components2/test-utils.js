import "../jest.setup";
import _ from 'lodash';
import {config, mount} from "@vue/test-utils";
import {hooks} from "./OrderView/pos-logic";
import {disablePay, payBtnClickable, payPrintMode, showIcon} from "./OrderView/pos-logic-be";
import StubFactory from './StubFactory';

jest.mock("vue-i18n", () => {
  return {
    useI18n() {
      return {
        t(t) {
          return t; // [Consider] return `t('${t}')` to indicate that i18n has been used
        },
        locale: "en"
      };
    }
  };
});
jest.mock("vue-router", () => {
  return {
    useRoute() {
      return {
        query: {
          type: 'default'
        }
      };
    },
    useRouter() {
      return {
        push() {
        }
      };
    }
  };
});

const dayjs = require("dayjs");
export const orm = require("schemahandler");
const syncFlow = require("schemahandler/sync/sync-flow");
const syncPlugin = require("schemahandler/sync/sync-plugin-multi");
const {stringify} = require("schemahandler/utils");
//jest.useFakeTimers("modern").setSystemTime(new Date("2021-01-01").getTime());
require("mockdate").set(new Date("2021-01-01").getTime());

const delay = require("delay");

export const foodTax = {taxes: [5, 10]};
export const drinkTax = {taxes: [16, 32]};

export const cola = {name: "Cola", price: 1.3, quantity: 1, ...drinkTax};
export const fanta = {name: "Fanta", price: 2, quantity: 1, ...drinkTax};
export const rice = {name: "Rice", price: 10, quantity: 1, ...foodTax};
export const ketchup = {name: "Add Ketchup", price: 3, quantity: 1};

beforeAll(async () => {
  orm.connect({uri: "mongodb://localhost:27017"}, "myproject");
  orm.registerSchema("Order", {
    inProgress: Boolean,
    items: [{}],
    table: Number
  });
  await orm("Order").deleteMany();
  await orm("Commit").deleteMany();

  orm.plugin(syncPlugin);
  orm.plugin(syncFlow);
  orm.registerCommitBaseCollection("Order");
  orm.plugin(require("../backend/commit/orderCommit"));
  orm.registerCollectionOptions("Order");
  orm.emit("commit:flow:setMaster", true);
});

afterEach(async () => {
  await orm("Order")
    .deleteMany()
    .direct();
  await orm("Commit")
    .deleteMany()
    .direct();
});

export let actions = [];
["showOrderReceipt", "pay", "printOrder"].forEach(e =>
  hooks.on(e, () => actions.push(e))
);
export let expectArray = () => [
  "payBtnClickable",
  payBtnClickable.value,
  "payPrintMode",
  payPrintMode.value,
  "showIcon",
  showIcon.value,
  "disablePay",
  disablePay.value,
  actions
];
//</editor-fold>

//<editor-fold desc="component">
export let component, wrapper;

export function setComponent(_component) {
  component = _component;
}

/**
 *
 * @param _component
 * @param options
 * @param useDefaults indicate whether _.defaults will be use for options
 * @returns {*}
 * example: makeWrapper(Order2, {shallow: false});
 */
export const makeWrapper = (_component, options, useDefaults = false) => {
  const defaultOption = {
    props: {},
    shallow: true,
    global: {
      directives: {
        touch: true
      },
      stubs: {
        'choose-table-dialog': true,
        'dialog-change-value': true,
        'dialog-choose-popup-modifier': true,
        "dialog-config-order-item": true,
        'dialog-form-input': true,
        'dialog-multi-payment': true,
        'dialog-multi-payment2': true,
        'dialog-product-search-result': true,
        'dialog-text-filter': true,
        'g-avatar': true,
        'g-autocomplete': true,
        'g-badge': true,
        'g-btn': true,
        "g-btn-bs": true,
        'g-card': true,
        'g-card-title': true,
        'g-combobox': true,
        'g-dialog': true,
        'g-divider': true,
        "g-icon": true,
        'g-item-group': true,
        'g-menu': true,
        "g-overlay": true,
        'g-radio': true,
        'g-simple-table': true,
        'g-spacer': true,
        'g-switch': true,
        'g-table': true,
        'g-tabs': true,
        'g-tab-item': true,
        'g-text-field': true,
        'g-text-field-bs': true,
        'g-toolbar': true,
        'g-snackbar': true,
        'pos-order': true,
        'pos-quick-order-toolbar': true,
        //'pos-order-split-order': true,
        'pos-keyboard-full': true,
        'pos-order-keyboard': true,
        'pos-order-move-items': true,
        'pos-order-receipt': true,
        'pos-order-voucher-dialog': true,
        'pos-textfield-new': true,
        'g-grid-select': StubFactory('g-grid-select'),
        'pos-payment-screen-payment-methods2': true,
        'pos-payment-screen-keyboard2': true,
        'pos-restaurant-payment-order-detail2': true,
        'pos-restaurant-payment-toolbar2': true,
      },
      mocks: {
        t: a => a,
        $t: a => a,
        $filters: {
          formatCurrency(val, decimals = 2) {
            if (!val || isNaN(val) || Math.floor(val) === val) return val;
            return val.toFixed(decimals);
          }
        }
      }
    }
  }
  const optionResult = useDefaults ? _.defaults(options, defaultOption) : _.merge(defaultOption, options)
  wrapper = mount(component || _component, optionResult);
};

beforeAll(() => (config.renderStubDefaultSlot = true));
