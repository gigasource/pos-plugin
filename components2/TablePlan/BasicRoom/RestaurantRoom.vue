<script>

import { roomUiFactory } from '../RoomUi'
import {
  initRouter,
  isTransferringTable,
  onCustomerDialogSubmit,
  showNumberOfCustomersDialog,
  chooseTable
} from './RestaurantRoomLogics'
import { getDiffTime } from '../../../utils/commons'
import { onActivated, onBeforeUnmount, onDeactivated, ref, watch } from 'vue';
import RoomStyleFactory from '../RoomStyles'
import { roomsFactory } from '../RoomState';
import { getTableOrderInfo, isBusyTable, isTable } from '../RoomShared';
import Touch from '../../../../../backoffice/pos-vue-framework/src/directives/touch/touch';
import { useI18n } from 'vue-i18n';
import RestaurantRoomEventHandlers from './EventHandlersForRestaurantRoom'
import { appHooks, user } from '../../AppSharedStates';
import NumberOfCustomersDialog from './NumberOfCustomersDialog';

export default {
  name: 'RestaurantRoom',
  components: { NumberOfCustomersDialog },
  props: {
    roomId: String
  },
  directives: { Touch },
  setup(props, { emit }) {
    // appHooks.emit('updateTseConfig')
    initRouter()
    onActivated(() => {
      console.log('room activated')
    })
    onDeactivated(() => {
      console.log('room deactivated')
    })
    const { t, locale } = useI18n()
    const { touchHandlers } = RestaurantRoomEventHandlers()
    const { selectingRoomStates, objectsInSelectingRoom } = roomsFactory(props);
    const { roomObjectContainerStyle, roomObjectStyle } = RoomStyleFactory(selectingRoomStates)
    const curTime = ref(new Date())

    const timerInterval = setInterval(() => curTime.value = new Date(), 30000)
    onBeforeUnmount(() => clearInterval(timerInterval));

    const objectContentRender = (obj) => {
      const isActiveTable = obj.type === 'table' && isBusyTable(obj)
      const order = getTableOrderInfo(obj)
      const zoom = selectingRoomStates.value ? selectingRoomStates.value.zoom : 1
      const isUserTable = (order, user) => {
        if (!user) return false
        //todo: check this logic: compare by name is correct ?
        return _.some(order.user, i => i.name === user.name)
      }
      return <div style={roomObjectStyle(obj)}>
        {isTable(obj) && <>
          {isBusyTable(obj) && isUserTable(order, user.value) &&
          <g-icon
              style="position: absolute; top: 0; right: 0; height: 13px; width: 15px">icon-room-border
          </g-icon>
          }
          {isBusyTable(obj) &&
          <div style={`font-size: ${11 * zoom}px ;position: absolute; top: 2px`}>
            {t('common.currency', locale.value)}{order.vSum}
          </div>}
          <div>
            <div> {obj.name} </div>
          </div>
          {isActiveTable ?
              <div style={`font-size: ${11 * zoom}px; position: absolute; bottom: 2px`}>
                {`${getDiffTime(order.date, curTime.value)} mins`} </div> : null}
        </>}
      </div>
    }

    const classes = (obj) => {
      let res = []
      res = isTable(obj) ? ['waves-effect', 'waves-red'] : []
      if (isTransferringTable(obj)) {
        res.push('animated', 'bounce', 'infinite')
      }
      return res
    }

    const _roomObjectContainerStyle = (obj) => {
      const style = roomObjectContainerStyle(obj)
      if (isBusyTable(obj)) Object.assign(style, { background: '#fec8c8', border: '1px solid #d2691e' })
      return style
    }

    const objectRender = (obj) => {
      return <div key={obj._id} id={obj.name}
                  style={_roomObjectContainerStyle(obj)}
                  class={classes(obj)}
                  onClick={() => chooseTable(obj)}
                  v-touch={touchHandlers(obj)}
      >
        {objectContentRender(obj)}
      </div>
    }

    const { renderRoom } = roomUiFactory(objectRender, selectingRoomStates, objectsInSelectingRoom);
    // const numberOfCustomersDialog = () =>
    //     <NumberOfCustomersDialog v-model={showNumberOfCustomersDialog.value}
    //                              onSubmit={({ numberOfCustomers, tseMethod }) => onCustomerDialogSubmit({ numberOfCustomers, tseMethod }, emit)}
    //     >
    //
    //     </NumberOfCustomersDialog>
    return () => <>
      {renderRoom()}

    </>
  }
}
</script>
<style scoped lang="scss">
.room {
  position: relative;
  width: 100%;
  height: 100%;
  background-image: url('/plugins/pos-plugin/assets/out.png');

  &__object__resizer {
    width: 24px;
    height: 24px;
    border-radius: 100%;
    background-color: #2979FF;
    position: absolute;
    right: 0;
    bottom: 0;
    transform: translate(100%, 100%);
    display: flex;
    justify-content: center;
    align-items: center;
  }

  &__object__rotate {
    width: 16px;
    height: 16px;
    border-radius: 100%;
    background-color: #2979FF;
    position: absolute;
    right: 0;
    bottom: 0;
    transform: translateX(100%);
    display: flex;
    justify-content: center;
    align-items: center;
  }
}
</style>
