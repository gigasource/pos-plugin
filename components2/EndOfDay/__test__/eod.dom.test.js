//<editor-fold desc="declare">
import {makeWrapper, orm, wrapper} from "../../test-utils";

import {nextTick} from "vue";
import {demoData} from "../../OrderView/__test__/demoData";
import RestaurantRoom from "../../TablePlan/BasicRoom/RestaurantRoom2";
import {fetchRooms, onSelectRoom, roomsStates} from "../../TablePlan/RoomState";
import EditableRoom from "../../TablePlan/EditableRoom/EditableRoom";
import PosEndOfDayDatePicker from "../PosEndOfDayDatePicker";

const {stringify} = require("schemahandler/utils");

const delay = require("delay");

//</editor-fold>

describe("eod test", function () {
  beforeAll(async () => {
    await orm("PosSetting").remove({});
    await orm("PosSetting").create(demoData.PosSetting[0]);
    await orm("OrderLayout").remove({});
    await orm("OrderLayout").create(demoData.OrderLayout[0]);
    await orm("Room").remove({});
    await orm("Room").create(demoData.Room);
  });

  it("case 1 date-picker", async function () {
    //order have 1 sent item, add one item -> should display print,
    //todo: fetch data
    //todo: convert singeton -> factory
    //room -> props
    //todo: dashboard: keep-alive
    await fetchRooms()
    makeWrapper(PosEndOfDayDatePicker, {
      shallow: true,
      props: {

      }
    });
    await nextTick();
    await delay(50);
    expect(wrapper.html()).toMatchSnapshot()
  }, 80000);


});
