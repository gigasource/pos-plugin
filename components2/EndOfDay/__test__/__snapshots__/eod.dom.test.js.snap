// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eod test case 1 date-picker 1`] = `
<g-picker-stub>
  <div class="g-date-picker-header">
    <div class="g-date-picker-header-container">
      <button class="g-date-picker-header__prev-button">
      </button>
      <div class="g-date-picker-header__value">
        <div>
          <button style="width: 208px;">
            January 2021
          </button>
        </div>
      </div>
      <button class="g-date-picker-header__next-button">
      </button>
    </div>
  </div>
  <div class="g-date-picker-table g-date-picker-table--date">
    <table>
      <thead>
        <tr>
          <th>
            S
          </th>
          <th>
            M
          </th>
          <th>
            T
          </th>
          <th>
            W
          </th>
          <th>
            T
          </th>
          <th>
            F
          </th>
          <th>
            S
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
          </td>
          <td>
          </td>
          <td>
          </td>
          <td>
          </td>
          <td>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                01
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                02
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                03
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                04
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                05
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                06
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                07
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                08
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                09
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                10
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                11
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                12
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                13
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                14
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                15
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                16
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                17
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                18
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                19
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                20
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                21
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                22
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                23
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                24
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                25
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                26
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                27
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                28
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                29
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                30
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <button type="button"
                    class="g-table-item"
            >
              <div class="g-table-item__content">
                31
              </div>
              <div class="g-date-picker-table__events">
              </div>
            </button>
            <div class="g-table-item__background">
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</g-picker-stub>
`;
